tests/core/data/testdataset.py

```diff
@@ -1551,8 +1557,8 @@ def test_dataset_2D_reduce_hm(self):
         array = da.from_array(np.random.rand(11, 11), 3)
         dataset = Dataset({'x':self.xs, 'y':self.y_ints, 'z': array},
                           kdims=['x', 'y'], vdims=['z'])
-        self.assertEqual(np.array(dataset.reduce(['x', 'y'], np.mean)),
-                         np.mean(array))
+        self.assertEqual((dataset.reduce(['x', 'y'], np.mean)),
+                         np.mean(array).compute())
```

Comment:
FWIW we have utility functions in Dask like [`assert_eq`]( https://github.com/dask/dask/blob/83c3129ce24f71e13e97219fb5c4da1b7d774668/dask/array/utils.py#L73 ). They aren't public technically (would like to change that personally). However they do a lot of things right, which might not be obvious initially.

Reply:
Thanks that's useful to know. I don't think they are needed in this case but I can think of other places where that would come in handy.